src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -764,49 +766,26 @@ mongoc_client_connect_unix (const mongoc_host_list_t *host, bson_error_t *error)
 #endif
 }
 
-
-/*
- *--------------------------------------------------------------------------
- *
- * mongoc_client_default_stream_initiator --
- *
- *       A mongoc_stream_initiator_t that will handle the various type
- *       of supported sockets by MongoDB including TCP and UNIX.
- *
- *       Language binding authors may want to implement an alternate
- *       version of this method to use their native stream format.
- *
- * Returns:
- *       A mongoc_stream_t if successful; otherwise NULL and @error is set.
- *
- * Side effects:
- *       @error is set if return value is NULL.
- *
- *--------------------------------------------------------------------------
- */
-
 mongoc_stream_t *
-mongoc_client_default_stream_initiator (const mongoc_uri_t *uri,
-                                        const mongoc_host_list_t *host,
-                                        void *user_data,
-                                        bson_error_t *error)
+mongoc_client_connect (bool buffered,
+                       bool use_ssl,
+                       void *ssl_opts_void,
```

Comment:
Sidenote: I found it frustrating to work with mongoc_ssl_opt_t. Since it is conditionally defined, using it requires either hiding the type or using typedefs.